e2e_test.ts

```diff
@@ -34,40 +33,47 @@ import options from "./fresh.config.ts";
  */
 const handler = await createHandler(manifest, options);
 
-function assertResponseNotFound(resp: Response) {
-  assertFalse(resp.ok);
-  assertEquals(resp.status, Status.NotFound);
+function assertHtml(resp: Response) {
+  assertInstanceOf(resp.body, ReadableStream);
+  assertEquals(
+    resp.headers.get("content-type"),
+    "text/html; charset=utf-8",
+  );
 }
 
-function assertResponseJson(resp: Response) {
-  assert(resp.ok);
-  assertNotEquals(resp.body, null);
+function assertJson(resp: Response) {
+  assertInstanceOf(resp.body, ReadableStream);
   assertEquals(resp.headers.get("content-type"), "application/json");
 }
 
-Deno.test("[e2e] GET /", async () => {
-  const resp = await handler(new Request("http://localhost"));
+function assertRedirect(response: Response, location?: string) {
+  assert(isRedirectStatus(response.status));
+  if (location !== undefined) {
+    assertStringIncludes(response.headers.get("location")!, location);
+  } else {
+    assert(response.headers.has("location"));
+  }
+}
 
-  assert(resp.ok);
+function assertXml(resp: Response) {
   assertInstanceOf(resp.body, ReadableStream);
   assertEquals(
     resp.headers.get("content-type"),
-    "text/html; charset=utf-8",
+    "application/atom+xml; charset=utf-8",
   );
-  assertEquals(resp.status, 200);
+}
+
+Deno.test("[e2e] GET /", async () => {
+  const resp = await handler(new Request("http://localhost"));
+
+  assertHtml(resp);
 });
 
 Deno.test("[e2e] GET /callback", async () => {
   const resp = await handler(
     new Request("http://localhost/callback"),
   );
 
-  assertFalse(resp.ok);
-  assertInstanceOf(resp.body, ReadableStream);
-  assertEquals(
-    resp.headers.get("content-type"),
-    "text/html; charset=utf-8",
-  );
   assertEquals(resp.status, 500);
```

Comment:
Let's make use of the `Status` enum where we can. It let's the reader know what the status code means.